src/libsyntax_pos/lib.rs

```diff
@@ -1258,6 +1286,12 @@ fn normalize_newlines(src: &mut String) {
         if tail.len() == gap_len {
             break;
         }
+        last_diff += 1;
+        cursor += idx - gap_len;
+        normalized_pos.push(NormalizedPos {
+            pos: BytePos::from_usize(cursor),
+            diff: last_diff
```

Comment:
`last_diff` and `gap_len` vary exactly the same, so let's do `diff: gap_len + original_diff` instead of maintaining an extra induction variable